source/client.c

```diff
@@ -851,6 +851,19 @@ int aws_mqtt_client_connection_connect(
             (uint64_t)connection_options->ping_timeout_ms, AWS_TIMESTAMP_MILLIS, AWS_TIMESTAMP_NANOS, NULL);
     }
 
+    /* Keep alive time should always be greater than the timeouts. */
+    if (AWS_UNLIKELY(
+            connection->keep_alive_time_secs * (uint64_t)AWS_TIMESTAMP_NANOS <= connection->request_timeout_ns)) {
+        AWS_LOGF_FATAL(
+            AWS_LS_MQTT_CLIENT,
+            "id=%p: Illegal configuration, Connection keep alive %" PRIu16
+            "s must be greater than the request timeouts %" PRIu64 "ns.",
+            (void *)connection,
+            connection->keep_alive_time_secs,
+            connection->request_timeout_ns);
+        AWS_FATAL_ASSERT(0);
```

Comment:
If we keep this, just assert the condition above, so that the assert is helpful.